File: Week9/ProblemA.c
/*
Language: c
ID: 63992
QLink: https://quera.org/course/assignments/63992/problems
Author: @Amirkhaskar
*/

#include <stdio.h>

void solve()
{
    int n;
    scanf("%d", &n);
    char c[n];
    scanf("%s", c);
    int i, count1 = 0, count2 = 0, count = 0;
    for (i = 0; i < n; i++)
    {
        if (c[i] == '(')
        {
            count1++;
        }
        else if (c[i] == ')')
        {
            count2++;
        }
        if (count2 > count1)
        {
            count++;
            count2 = count1;
        }
    }
    printf("%d", count);
}
int main()
{
    int t;
    scanf("%d", &t);
    while (t--)
    {
        solve();
    }
    return 0;
}

File: Week9/ProblemA.py
# Language: python
# ID: 63992
# QLink: https://quera.org/course/assignments/63992/problems
# Author: @Amirkhaskar

for t in range(int(input())):
	n=input()
	s=input()
	while '()' in s:
		s=s.replace('()','')
	print(len(s)//2)